### PR TITLE
Created the route for campaign deletion

### LINKED ISSUE #135: Delete Campaign
# Issue Template

## Issue Type

- [ ] Bug Report
- [ ] Feature Request
- [ ] Enhancement
- [ ] Other (Please specify)

## Description

Allow deletion of created campaign if it has not been published / has not received donations yet!

**Note:** Please make sure to follow the project's code of conduct and contribution guidelines when creating an issue. Thank you for contributing to our open-source project!

### PR DESCRIPTION
**Pull request by Davheed2**
# Pull Request Template

## Created the route for campaign deletion

Please include a summary of the change and which issue is fixed (if applicable).

## Related Issue

- Fixes #135 

## Checklist

- [x] I have reviewed the Contribution Guidelines linked above.
- [x] I have tested my changes thoroughly and ensured that all existing tests pass.
- [x] I have provided clear and concise commit messages.
- [ ] I have updated the project's documentation as necessary.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
 		"start": "node -r ts-node/register/transpile-only -r tsconfig-paths/register build/server.js",
 		"start-prod": "node build/server.js",
 		"build": "tsc",
-		"dev": "./start-dev-server.sh",
+		"dev": "bash start-dev-server.sh",
```
Comment (1finedev) [inline package.json:12]: why was this chanage made?

File: src/common/interfaces/campaign.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ interface ICampaign {
 	image: string;
 	title: string;
 	deadline: Date;
+	deletedDate: Date;
```
Comment (1finedev) [inline src/common/interfaces/campaign.ts:12]: add a boolean to this model, isDeleted: true, alongside the deletion date to enable queries easily ignore deleted campaigns

File: src/controllers/campaign/delete.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import AppError from '@/common/utils/appError';
+import { catchAsync } from '@/middlewares';
+import { Request, Response } from 'express';
+import { campaignModel as campaign } from '@/models/campaignModel';
+import { DateTime } from 'luxon';
+
+const deleteCampaign = catchAsync(async (req: Request, res: Response) => {
+	const { user } = req;
+
+	if (!user) {
```
Comment (1finedev) [inline src/controllers/campaign/delete.ts:11]: this check is not required as this is a protected route and there will always be a user

File: src/controllers/campaign/delete.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		throw new AppError('User not found!. Please login');
+	}
+
+	const { campaignId } = req.params;
+	const updatedCampaign = await campaign.findByIdAndUpdate(campaignId, {
```
Comment (1finedev) [inline src/controllers/campaign/delete.ts:16]: this implementation allows for any authenticated user to delete another person's campaign provided they know the campaign id, add a check to ensure the currently logged in user ID matches the campaign creator user ID

File: src/controllers/campaign/stepTwo.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { AppResponse } from '@/common/utils';
+import AppError from '@/common/utils/appError';
+import { catchAsync } from '@/middlewares';
+import { campaignModel as campaign } from '@/models/campaignModel';
+import { Request, Response } from 'express';
+
+const secondStep = catchAsync(async (req: Request, res: Response) => {
+	const { user } = req;
+	const { title, fundraiser, goal } = req.body;
+	const { campaignId } = req.params;
```
Comment (1finedev) [inline src/controllers/campaign/stepTwo.ts:10]: Campaign ID will be in the req body not in params, only step would be be sent as param, also verify that the campaign to be updated was created by the logged in user!

File: src/controllers/campaign/stepTwo.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+	const currentDateDeadline = Date.now();
+
+	const step2Campaign = await campaign.findByIdAndUpdate(
+		campaignId,
+		{
+			title,
+			fundraiser,
+			goal,
+			deadline: currentDateDeadline,
```
Comment (1finedev) [inline src/controllers/campaign/stepTwo.ts:28]: using the Date.now() means the campaign will expire right now?? why is this so

File: package.json (hunk 1 above)
Comment (JC-Coder) [reply package.json:12]: windows not able to run the command withouth the bash initializer.

File: src/controllers/campaign/stepTwo.ts (hunk 6 above)
Comment (Davheed2) [reply src/controllers/campaign/stepTwo.ts:28]: the Date.now(),  was just added as an extra measure to make sure the campaign was created at that particular time. Removing it if not needed

**Pull request closed without merging**